packages/wdio-static-server-service/src/launcher.ts

```diff
@@ -1,45 +1,50 @@
-import path from 'path'
+import { join, resolve } from 'path'
 import { promisify } from 'util'
-
 import express from 'express'
 import fs from 'fs-extra'
 import morgan from 'morgan'
 import logger from '@wdio/logger'
+import { FolderOption, MiddleWareOption } from './types'
 
 const log = logger('@wdio/static-server-service')
 
 const DEFAULT_LOG_NAME = 'wdio-static-server-service.log'
 
 export default class StaticServerLauncher {
-    constructor ({ folders, port = 4567, middleware = [] }) {
+    folders: FolderOption[] | null
+    port: number
+    middleware: MiddleWareOption[]
+    server!: express.Express
+    constructor({ folders, port = 4567, middleware = [] }: { folders?: FolderOption[] | FolderOption, port?: number, middleware?: MiddleWareOption[] }) {
         this.folders = folders ? Array.isArray(folders) ? folders : [folders] : null
         this.port = port
         this.middleware = middleware
     }
 
-    async onPrepare ({ outputDir }) {
+    async onPrepare({ outputDir }: { outputDir?: string }) {
         if (!this.folders) {
             return
         }
 
         this.server = express()
 
         if (outputDir) {
-            const file = path.join(outputDir, DEFAULT_LOG_NAME)
+            const file = join(outputDir, DEFAULT_LOG_NAME)
             fs.createFileSync(file)
             const stream = fs.createWriteStream(file)
             this.server.use(morgan('tiny', { stream }))
         }
 
-        this.folders.forEach((folder) => {
-            log.info('Mounting folder `%s` at `%s`', path.resolve(folder.path), folder.mount)
+        this.folders.forEach((folder: FolderOption) => {
+            log.info('Mounting folder `%s` at `%s`', resolve(folder.path), folder.mount)
             this.server.use(folder.mount, express.static(folder.path))
         })
 
         this.middleware.forEach(
-            (ware) => this.server.use(ware.mount, ware.middleware))
+            (ware: MiddleWareOption) => this.server.use(ware.mount, ware.middleware as unknown as express.Application))
 
-        await promisify(this.server.listen.bind(this.server))(this.port)
+        const listen: (port: number) => Promise<any> = promisify(this.server.listen)
```

Comment:
this should fix the issue.
```suggestion
        const listen = promisify(this.server.listen)
```

Reply:
If i do the above changes then i am getting the below error:

```
(property) StaticServerLauncher.port: number
Expected 0 arguments, but got 1
```

Reply:
using this ` const listen = <(port: number) => Promise<any>> promisify(this.server.listen)`. is this Ok?